urlc8dbc7ce6e_dlubarov_guava_tgz-pJ8-MainJ8/src/c/MethodDef.java

```diff
@@ -0,0 +1,240 @@
+package c;
+
+import java.util.*;
+
+import common.*;
+
+import c.ty.*;
+import c.gen.MethodGenericInfo;
+import c.stm.*;
+
+public class MethodDef {
+    public final RawType owner;
+
+    public final MethodVisibility visibility;
+    public final boolean isStatic, isSealed;
+    public final Type returnType;
+    public final String name;
+    public final MethodGenericInfo[] genericInfos;
+    public final Type[] paramTypes;
+    public final String[] paramNames;
+    public final Block body;
+
+    private final List<RawType> rawTypeTable;
+    private final List<Type> fullTypeTable;
+    private final List<MethodDef> methodTable;
+    private final List<String> stringTable;
+
+    private int mostLiveLocals = 0;
+
+    public MethodDef(RawType owner,
+            MethodVisibility visibility, boolean isStatic, boolean isSealed,
+            Type returnType, String name, MethodGenericInfo[] genericInfos,
+            Type[] paramTypes, String[] paramNames, Block body) {
+        this.owner = owner;
+        this.visibility = visibility;
+        this.isStatic = isStatic;
+        this.isSealed = isSealed;
+        this.returnType = returnType;
+        this.name = name;
+        this.genericInfos = genericInfos;
+        this.paramTypes = paramTypes;
+        this.paramNames = paramNames;
+        this.body = body;
+
+        rawTypeTable = new ArrayList<RawType>();
+        fullTypeTable = new ArrayList<Type>();
+        methodTable = new ArrayList<MethodDef>();
+        stringTable = new ArrayList<String>();
+
+        if (isStatic && TypeUtils.containsTypeParams(paramTypes))
+            throw new NiftyException("Static method '%s.%s' takes type parameter.", owner, name);
+    }
+
+    public boolean canImplement(MethodDef that) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 15. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 77-99 into a separate method.